app/src/features/sources/SourceCreate.tsx

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+
+import { Button, makeStyles, Typography } from "@material-ui/core";
+
+import { useAppDispatch } from "../../app/store";
+import { Source } from "../../services/api/generated/api.generated";
+import { editSource } from "./sourceSlice";
+
+const useStyles = makeStyles(() => ({
+  button: {
+    textTransform: "none",
+  },
+}));
+
+const SourceCreate = (): JSX.Element => {
+  const classes = useStyles();
+  const dispatch = useAppDispatch();
+
+  const handleClick = () => dispatch(editSource({} as Source));
```

Comment:
Casting `{}` as a Source disturbs me a little..
Can we call `editSource` with `null` instead to mean that our source is to be created ? I don't know if that would be better practice

Reply:
What I don't really like is casting it as a source when `name` is not set even though it is required

Reply:
I'll add a `name` attribute (with an empty string)! The idea here is to say that we want to edit a new Source so we initialize a new one which open the drawer (in `SourceForm`)